rclcpp/test/rclcpp/test_generic_pubsub.cpp

```diff
@@ -190,7 +190,7 @@ TEST_F(RclcppGenericNodeFixture, publish_loaned_msg_work)
 
   if (publisher->can_loan_messages()) {
     auto subscriber_future_ = std::async(
-      std::launch::deferred, [this, topic_name, type] {
+      std::launch::async, [this, topic_name, type] {
```

Comment:
The test was never enabled before https://github.com/ros2/rmw_fastrtps/pull/568. Instead of this PR, this test fails with TIMEOUT(2).

Reply:
could you explain why is the change from deferred to async needed?